File: Implementation/python/15_문자열재정렬.py
# 이것이 취업을 위한 코딩테스트다.
# 
# 1

S = list(input())
result = []
cnt = 0

for c in S:
    if 'A' <= c and c <= 'Z':
        result.append(c)
    else:
        cnt += int(c)
result.sort()
if cnt != 0:
    result.append(str(cnt))
print(''.join(result))

File: backup/Implementation/c++/15_문자열재정렬.cpp
// 이것이 취업을 위한 코딩테스트다 322p
//
// 2

#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

int main()
{
	string s, result = "";
	cin >> s;

	int sum = 0;
	for (int i = 0; i < s.size(); i++) {
		if ('0' <= s[i] && s[i] <= '9')
			sum += s[i] - '0';
		else
			result += s[i];
	}
	sort(result.begin(), result.end());
	if (sum != 0)
		result += to_string(sum);
	cout << result;
}